### PR TITLE
Add Prism Launcher (7.0) and dependencies

### PR DESCRIPTION
**Pull request by anellie**
This PR adds PrismLauncher, along with 2 build dependencies it needed (qt6-5compat, extra-cmake-modules).
This is my first PR, so I hope I got the templates right, especially since I wasn't sure of a good way to specify the Java installation for PL.
It builds and runs fine on x86; it should too on ARM and PPC but I was unable to test it due to not having any hardware available.

### CODE REVIEW

**Comment by q66** [comment]
make sure to run your files through `black` and `flake8` ideally beforehand, right now it's not so important as the CI is not loaded significantly, but once there is more stuff it may result in unnecessary stalls

**Comment by anellie** [comment]
Yeah, will do next time! I admittedly just forgot with these, will fix the formatting and redo the commits.

**Comment by anellie** [comment]
Should be good to go now! `flake8 contrib` reports no errors. Sorry for the trouble and thanks for the review!

File: contrib/extra-cmake-modules/template.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+pkgname = "extra-cmake-modules"
+pkgver = "5.107.0"
+pkgrel = 0
+build_style = "cmake"
+hostmakedepends = ["cmake", "ninja"]
+makedepends = []
```
Comment (q66) [inline contrib/extra-cmake-modules/template.py:6]: you can remove this one

File: contrib/extra-cmake-modules/template.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+pkgdesc = "Extra modules and scripts for CMake"
+maintainer = "aurelia <git@elia.garden>"
+license = "BSD-3-Clause"
+url = "https://github.com/KDE/extra-cmake-modules"
+source = f"{url}/archive/v{pkgver}.tar.gz"
+sha256 = "b59d1bd4723be532ada21c1e584e2e61940e066f7ace0131173095431700bec3"
+
```
Comment (q66) [inline contrib/extra-cmake-modules/template.py:13]: no need for the empty line here

File: contrib/prismlauncher/template.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+pkgname = "prismlauncher"
+pkgver = "7.0"
+pkgrel = 0
+build_style = "cmake"
+# TODO is this really the ideal way to do this?
+configure_env = {"JAVA_HOME": "/usr/lib/jvm/java-17-openjdk"}
```
Comment (q66) [inline contrib/prismlauncher/template.py:6]: i'll think about this one a bit more if you don't mind

File: contrib/qt6-5compat/template.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+makedepends = ["qt6-qtbase-devel"]
+pkgdesc = "Module containing unsuppored Qt5 APIs"
+maintainer = "aurelia <git@elia.garden>"
+license = (
+    "LGPL-2.1-only AND LGPL-3.0-only AND GPL-3.0-only WITH Qt-GPL-exception-1.0"
+)
+url = "https://www.qt.io"
+source = f"https://download.qt.io/official_releases/qt/{pkgver[:-2]}/{pkgver}/submodules/qt5compat-everywhere-src-{pkgver}.tar.xz"
+sha256 = "a9e2f53a193fc2e131b01a2f6e7a1fbfe31309c2413fdc213e5a81c558c21261"
+
```
Comment (q66) [inline contrib/qt6-5compat/template.py:15]: ditto empty line shouldn't be here

**Review by q66 — CHANGES_REQUESTED** [review]
looks good otherwise, just please wait a bit until i address the _JAVA_HOME stuff :)

**Comment by anellie** [comment]
Redid the commits with the requested changes and removed the `JAVA_HOME`, which `java-common`  should handle now. Have not built it yet due to r1 of the JDK not being available as a binary package yet

**Comment by anellie** [comment]
Weird, it still behaves the same as it did without specifying `JAVA_HOME`... `java_common` should have taken care of doing so, correct?

**Comment by q66** [comment]
profile.d is for login shells, package builds don't spawn a login shell

**Comment by q66** [comment]
well, just do the configure_env, there is not much point in setting up anything more elaborate

**Comment by q66** [comment]
qt6-5compat should be called qt6-qt5compat to follow the existing local and upstream convention, otherwise lgtm

**Comment by anellie** [comment]
Renamed it now

**Pull request merged**